sinks/cache/cache.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import source_api "github.com/GoogleCloudPlatform/heapster/sources/api"
+
+type Metadata struct {
+	Name      string
+	Namespace string
+	UID       string
+	Hostname  string
+	Labels    map[string]string
+}
+
+type ContainerElem struct {
+	Metadata
+	ContainerSpec  TimeStore
+	ContainerStats TimeStore
+}
+
+type PodElem struct {
+	Metadata
+	// map of container name to container element.
+	Containers map[string]*ContainerElem
+	// TODO: Cache history of Spec and Status.
+}
+
+type NodeElem struct {
+	Node *ContainerElem
+	// FreeContainers refers to all the containers in a node
+	// that do not belong to a pod.
+	FreeContainers map[string]*ContainerElem
```

Comment:
Not a huge fan of that either, ExternalContainers was another suggestion but that was shut down. It doesn't matter as long as we're consistent though :)